lib/FieldType/Mapper/AuthorFormMapper.php

```diff
@@ -60,16 +63,14 @@ public function mapFieldValueForm(FormInterface $fieldForm, FieldData $data)
         $fieldSettings = $fieldDefinition->getFieldSettings();
         $formConfig = $fieldForm->getConfig();
         $names = $fieldDefinition->getNames();
-        $label = $fieldDefinition->getName($formConfig->getOption('languageCode'))
```

Comment:
Uh :)
Actually, assuming you've obtained that `FieldDefinition` from a Content Type loaded via `loadContentType($contentTypeId, $languageCodes);`, where `$languageCode` comes from `UserLanguagePreferenceProvider`, the call to `$fieldDefinition->getName()` should get you a proper label. 

This code should not have been here in the first place, because it repeats internal Repository's logic.